crates/metadata-server/src/local/server.rs

```diff
@@ -152,22 +152,88 @@ impl LocalMetadataServer {
+
         if let Some(node_config) =
             nodes_configuration.find_node_by_name(Configuration::pinned().common.node_name())
         {
-            if matches!(
+            // Only needed if we resume from a Restate version that has not properly set the
+            // MetadataServerState to Member in the NodesConfiguration.
+            if !matches!(
                 node_config.metadata_server_config.metadata_server_state,
                 MetadataServerState::Member
             ) {
-                // nothing to patch
-                return Ok(());
+                info!(
+                    "Setting MetadataServerState to Member in NodesConfiguration for node '{}'",
+                    node_config.name
+                );
+                let mut new_node_config = node_config.clone();
+                new_node_config.metadata_server_config.metadata_server_state =
+                    MetadataServerState::Member;
+
+                nodes_configuration.upsert_node(new_node_config);
+                modified = true;
             }
+        }
 
-            let mut new_node_config = node_config.clone();
-            new_node_config.metadata_server_config.metadata_server_state =
-                MetadataServerState::Member;
+        // If we have a node-id 0 in our NodesConfiguration, then we need to migrate it to another
+        // value since node-id 0 is a reserved value now. This can only be problematic if users are
+        // running multi node Restate cluster where there is a metadata-server node and others. In
+        // this situation if one of the other nodes was started with a forced node-id 0, then one
+        // needs to either unset this option or set it to the node-id that the migration step
+        // decided.
+        let zero = PlainNodeId::new(0);
+        if let Ok(node_config) = nodes_configuration.find_node_by_id(zero) {
+            let plain_node_id_to_migrate_to = if node_config.name
+                == Configuration::pinned().node_name()
+            {
+                if let Some(force_node_id) = Configuration::pinned().common.force_node_id {
+                    assert_ne!(
+                        force_node_id, zero,
+                        "It should no longer be allowed to force the node id to 0"
+                    );
+
+                    if let Ok(force_node_id_node_config) =
+                        nodes_configuration.find_node_by_id(force_node_id)
+                    {
+                        return Err(MigrationError::OccupiedNodeId(
+                            force_node_id,
+                            force_node_id_node_config.name.to_owned(),
+                        ));
+                    }
+
+                    force_node_id
+                } else {
+                    nodes_configuration
+                        .max_plain_node_id()
+                        .map(|node_id| node_id.next())
+                        .unwrap_or(PlainNodeId::MIN_PLAIN_NODE_ID)
+                }
```

Comment:
I'm really not sure if we need this if we assume that configs have a single node only. 0 => 1, that's it.

Reply:
It is overly protective, agreed. I don't expect people to run a multi-node setup (one metadata node and one worker node) with version < 1.2. So I am fine with simplifying the check.